src/components/Form.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+
+export default function Form(): JSX.Element {
+  const classes = useStyles();
+
+  return (
+    <Container component="main" maxWidth="xs">
+      <div className={classes.paper}>
+        <Typography component="h1" variant="h5">
+          Sign up
+        </Typography>
+        <form className={classes.form} noValidate>
+          <Grid container spacing={2}>
+            <Grid item xs={12} sm={6}>
+              <TextField
+                autoComplete="fname"
+                name="firstName"
+                variant="outlined"
+                required
+                fullWidth
+                id="firstName"
+                label="First Name"
+                autoFocus
+              />
+            </Grid>
+            <Grid item xs={12} sm={6}>
+              <TextField
+                variant="outlined"
+                required
+                fullWidth
+                id="lastName"
+                label="Last Name"
+                name="lastName"
+                autoComplete="lname"
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <TextField
+                variant="outlined"
+                required
+                fullWidth
+                id="email"
+                label="Email Address"
+                name="email"
+                autoComplete="email"
+              />
+            </Grid>
+            <Grid item xs={12}>
+              <TextField
+                variant="outlined"
+                required
+                fullWidth
+                name="password"
+                label="Password"
+                type="password"
+                id="password"
+                autoComplete="current-password"
+              />
+            </Grid>
+          </Grid>
+          <Button
+            type="submit"
+            fullWidth
+            variant="contained"
+            color="secondary"
+            className={classes.submit}
+          >
+            Sign Up
+          </Button>
+        </form>
+      </div>
+      <Box mt={5}></Box>
+    </Container>
+  );
+}
+
+const useStyles = makeStyles(function (theme) {
```

Comment:
A clarification on my "prefer functions" comment. I meant to prefer functions when declaring a component. I also prefer functions when declaring a stand-alone function. I am personally not a fan of

`const foo = (x, y) => { return doSomething(); };`

I find it difficult to read. However, when passing callbacks, I greatly prefer =>. So I would write this as

`makeStyles((theme) => { return doSomething(); })`

In fact, if at all possible, I prefer leaving out the {}. So I'd actually write it as:

`makeStyles((theme) => doSomething());`

I only put braces in when the syntax requires it. In this case, there's no need for them or the return statement.

Reply:
Thanks, noted and updated!